valory-xyz/agent-academy-2
# Get to simple

goal is to get to a functional system - some short cuts are taken in order to achieve this. Interacting with the [TestJob contract](https://goerli.etherscan.io/address/0xd50345ca88e0B2cF9a6f5eD29C1F1f9d76A16C3c#code) on Goerli

don't use current pytest agent setup to attempt to deploy, [they get robbed as soon as they receive funds](https://goerli.etherscan.io/address/0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266)

to test functionality
1. either 
   a. remove safe contract from `aea-config.yaml`. You will need some amount of gETH for gnosis safe contract deployment. Make sure to save the `ethereum_private_key.txt` somewhere so you can reuse it to access the safe next time around.
   b. ask me for the private key of [the agent](https://goerli.etherscan.io/address/0x1B621c19C3E868A4DF2E1858b08cedA8633927EA) used to deploy [this safe](https://goerli.etherscan.io/address/0xb3c1Cf4eeF9e3C2d165C75891DD24a494C54883c).
2. (personal) infura RPC end-point must be set in `aea-config.yaml` - ideally using environmental variable. 
   `export CONNECTION_LEDGER_CONFIG_LEDGER_APIS_ETHEREUM_ADDRESS=https://goerli.infura.io/v3/<my_infura_endpoint>`

then use CLI tool
```
aea fetch keep3r_co/keep3r_bot --local
cd keep3r_bot
aea generate-key ethereum
aea add-key ethereum
cp ethereum_private_key.txt <my_local_storage_location>
aea issue-certificates --aev
aea run --aev
```
